langchain4j-redis/src/main/java/dev/langchain4j/store/embedding/redis/RedisEmbeddingStore.java

```diff
@@ -314,20 +317,29 @@ public Builder dimension(Integer dimension) {
          */
         @Deprecated
         public Builder metadataFieldsName(Collection<String> metadataFieldsName) {
-            this.metadataKeys = metadataFieldsName;
-            return this;
+            return metadataKeys(metadataFieldsName);
         }
 
         /**
          * @param metadataKeys Metadata keys that should be persisted (optional)
+         * @deprecated use {@link #metadataConfig(Map)}} instead
          */
+        @Deprecated
```

Comment:
Should we really deprecate this? What if users are ok with default metadata config?

Reply:
Currently I implement this deprecated method by default (like previous set all metadata field as `TextField`). Actually I don't have plan to remove it right now, but using `metadataConfig()` will be more concrete and can use filter then.

Reply:
My concern is that by deprecating this method we will be forcing users to explicitly specify types for all metadata keys. I am not sure if it is good or not, WDYT? If user just wants to store metadata values, they should be able to do so without the need to specify types explicitly, or?

Reply:
hmmm, you are right. What about splitting it into like `tagFields(Collection<String> keys)`, `textFields(Collection<String> keys)` and so on. Thus users can just specify the metadata keys.

If users want to customize the Field Setting, the should call `metadataConfig` and pass types.

In ctor, we can merge all fields passed by `metadataConfig`, `tagFields`, `textFields` and so on, and then deduplicate it by name. (priority: metadataConfig > others).

WDYT?

Reply:
Sorry, but what are `tagFields`?

Reply:
Tag type is a field type represents shorter string. see [Redis Search field and type options](https://redis.io/docs/latest/develop/interact/search-and-query/basic-constructs/field-and-type-options/).

What I mean is that represent convinient builder method for all type, and take `Collection<String> keys` as param. Users can just pass the name of keys and don't need to specify field type. Some field type may has some advaned setting (such as `TextField` has `weight` param), users can specify it by passing `metadataConfig`.

Reply:
Sorry for delays, having too much on my plate lately, I will try to review it asap

Reply:
Thank you so much!